### PR TITLE
Feature/ex 7542 use highlight component in suggestions

### PR DESCRIPTION
**Pull request by javieri-empathy**
* Uses the `Highlight` component inside `BaseSuggestion`
* Re-dos test so they use the render function pattern

BREAKING-CHANGE: Removes `queryHTML` slot scope prop from `BaseSuggestion`, `PopularSearch` `QuerySuggestion`, `HistoryQuery`. Use new `Highlight` component instead.

### CODE REVIEW

File: packages/x-components/src/x-modules/history-queries/components/__tests__/history-query.spec.ts [hunk 1]
```diff
@@ -1,108 +1,163 @@
...
   });
 
-  it('button ClearHistoryQuery emits UserPressedRemoveHistoryQuery when it is clicked', () => {
-    const listener = jest.fn();
-    const removeHistoryQueryWrapper = historyQueryWrapper.find(
-      getDataTestSelector('remove-history-query')
-    );
-    removeHistoryQueryWrapper.vm.$x.on('UserPressedRemoveHistoryQuery', true).subscribe(listener);
-    removeHistoryQueryWrapper.trigger('click');
+  it('emits appropriate events on click', async () => {
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/history-queries/components/__tests__/history-query.spec.ts:81]:
```suggestion
  it('emits appropriate events on click', () => {
```

File: packages/x-components/src/x-modules/history-queries/components/__tests__/history-query.spec.ts [hunk 2]
```diff
@@ -1,108 +1,163 @@
...
+    });
 
-    expect(listener).toHaveBeenCalledTimes(1);
-    expect(listener).toHaveBeenCalledWith({
-      eventPayload: historyQuery,
-      metadata: {
-        moduleName: 'historyQueries',
-        target: removeHistoryQueryWrapper.element
-      }
+    await getSuggestionWrapper().trigger('click');
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/history-queries/components/__tests__/history-query.spec.ts:86]:
```suggestion
    getSuggestionWrapper().trigger('click');
```

File: packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts [hunk 3]
```diff
@@ -1,71 +1,115 @@
...
+      return wrapper.get(getDataTestSelector('matching-part'));
+    }
+  };
+}
 
-  it('is an XComponent', () => {
-    expect(isXComponent(popularSearchWrapper.vm)).toEqual(true);
-    expect(getXComponentXModuleName(popularSearchWrapper.vm)).toBe('popularSearches');
+describe('testing popular-search component', () => {
+  it('is an XComponent that belongs to the query suggestions', () => {
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts:49]:
```suggestion
  it('is an XComponent that belongs to the popular searches', () => {
```

File: packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts [hunk 4]
```diff
@@ -1,71 +1,115 @@
...
+    const { wrapper } = renderPopularSearch({
+      suggestion: createPopularSearch('milk')
     });
+    expect(wrapper.text()).toEqual('milk');
   });
 
-  it('renders a button with the query of the popular search (suggestion)', () => {
-    expect(popularSearchWrapper.find('span').element).toBeFalsy();
-    expect(popularSearchWrapper.element.innerHTML).toEqual(suggestion.query);
+  it('emits appropriate events on click', async () => {
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts:62]:
```suggestion
  it('emits appropriate events on click', () => {
```

File: packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts [hunk 5]
```diff
@@ -1,71 +1,115 @@
...
+    const { wrapper, emitSpy, suggestion } = renderPopularSearch({
+      suggestion: createPopularSearch('milk')
+    });
+
+    await wrapper.trigger('click');
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts:67]:
```suggestion
    wrapper.trigger('click');
```

File: packages/x-components/src/x-modules/query-suggestions/components/__tests__/query-suggestion.spec.ts [hunk 6]
```diff
@@ -1,67 +1,130 @@
...
 
-  // TODO: Refactor state to normalized query getter
-  it('highlights the suggestion matching parts with the state query', async () => {
-    resetXQuerySuggestionsStateWith(store, { query: 'Bá' });
+    expect(getMatchingPart().text()).toEqual('ba');
+    expect(wrapper.text()).toEqual('baileys');
+  });
 
-    await localVue.nextTick();
+  it('emits appropriate events on click', async () => {
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/query-suggestions/components/__tests__/query-suggestion.spec.ts:75]:
```suggestion
  it('emits appropriate events on click', () => {
```

File: packages/x-components/src/x-modules/query-suggestions/components/__tests__/query-suggestion.spec.ts [hunk 7]
```diff
@@ -1,67 +1,130 @@
...
+    const { wrapper, emitSpy, suggestion } = renderQuerySuggestion({
+      suggestion: createQuerySuggestion('milk')
+    });
 
-    expect(component.classes()).toContain('x-suggestion--matching');
-  });
+    await wrapper.trigger('click');
```
Comment (ajperezbau) [inline packages/x-components/src/x-modules/query-suggestions/components/__tests__/query-suggestion.spec.ts:80]:
```suggestion
   wrapper.trigger('click');
```

**Review by ajperezbau — CHANGES_REQUESTED** [review]
Good job!

File: packages/x-components/src/x-modules/history-queries/components/history-query.vue [hunk 8]
```diff
@@ -129,9 +127,14 @@ that serves to remove this query from the history. This slot only has one proper
 ```vue live
 <template>
   <HistoryQuery :suggestion="suggestion">
-    <template #default="{ suggestion, queryHTML }">
+    <template #default="{ suggestion, start, match, end, hasMatch, text }">
```
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/history-queries/components/history-query.vue:130]: Nope, and should not be bound. I'll remove them

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts [hunk 9]
```diff
@@ -1,71 +1,115 @@
...
+      store
+    }
+  );
 
-  const popularSearchWrapper = mount(PopularSearch, { localVue, propsData: { suggestion } });
+  return {
+    wrapper: wrapper.findComponent(PopularSearch),
+    suggestion,
+    emitSpy: jest.spyOn(XPlugin.bus, 'emit'),
+    getMatchingPart() {
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/popular-searches/components/__tests__/popular-search.spec.ts:42]: no needed here, right?

File: packages/x-components/src/x-modules/popular-searches/components/popular-search.vue [hunk 10]
```diff
@@ -5,14 +5,15 @@
...
-      -->
-      <slot :suggestion="suggestion">{{ suggestion.query }}</slot>
-    </template>
+    <!-- eslint-disable max-len -->
+    <!--
+        @slot Popular Search content
+          @binding {Object} v-bind - `BaseSuggestion` default slot scope: **suggestion** <code>Suggestion</code> - Suggestion data<br /> **query** <code>string</code> - The query that the suggestion belongs to<br /> **filter** <code>Filter \| undefined</code> - Suggestion's filter
+    -->
+    <!-- eslint-enable max-len -->
+    <slot v-bind="{ ...baseScope }" />
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/popular-searches/components/popular-search.vue:16]: necessary to create a copy?

File: packages/x-components/src/x-modules/query-suggestions/components/query-suggestion.vue [hunk 11]
```diff
@@ -1,21 +1,18 @@
...
-      <!-- eslint-enable max-len -->
-      <slot v-bind="{ suggestion, queryHTML }" />
-    </template>
+    <!-- eslint-disable max-len -->
+    <!--
+        @slot Query Suggestion content
+          @binding {Object} v-bind - `BaseSuggestion` default slot scope: **suggestion** <code>Suggestion</code> - Suggestion data<br /> **query** <code>string</code> - The query that the suggestion belongs to<br /> **filter** <code>Filter \| undefined</code> - Suggestion's filter
+    -->
+    <!-- eslint-enable max-len -->
+    <slot v-bind="{ ...baseScope }" />
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/query-suggestions/components/query-suggestion.vue:15]: necessary to create a copy?

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/popular-searches/components/popular-search.vue (hunk 10 above)
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/popular-searches/components/popular-search.vue:16]: Sadly yes, if I don't create a copy the slot docs are not generated

Comment (tajespasarela) [reply packages/x-components/src/x-modules/popular-searches/components/popular-search.vue:16]: WTF!

Comment (javieri-empathy) [reply packages/x-components/src/x-modules/popular-searches/components/popular-search.vue:16]: Proof:
image
image

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**